helix-term/src/ui/editor.rs

```diff
@@ -275,21 +275,40 @@ impl EditorView {
                 let end = text.line_to_char(last_line + 1);
                 Range::new(start, end)
             };
-            let cursor_style = Style::default()
-                // .bg(Color::Rgb(255, 255, 255))
-                .add_modifier(Modifier::REVERSED);
+            let scope = match doc.mode() {
+                Mode::Insert => "ui.cursor.insert",
+                Mode::Select => "ui.cursor.select",
+                Mode::Normal => "ui.cursor",
```

Comment:
Why shouldn't this be `ui.cursor.normal`?

Reply:
Was thinking that too at first. I would actually keep it as `ui.cursor`, this way the insert and select ones can fall back to it if they're not defined

Reply:
I was also considering making it `ui.cursor.normal` but I think as @archseer said it makes more sense as a fallback.

Reply:
But the issue is users would not understand the heuristic if all of us here "think that too at first". Then users have to go through our thought process to know why was the decision made this way. Do you think that is a better tradeoff for visibility vs simplicity?

Reply:
I didn't say I considered it first, I just said I considered it as an option. Actually at first I came up with `ui.cursor` because that is what for example vim does, you have `Cursor` and `iCursor` as highlight groups and not `nCursor`.

Reply:
I think it makes sense, you're defining a color for `ui.cursor` but if you'd like to change the style for `ui.cursor.insert` or `ui.cursor.select` you can override it. I anticipate most themes should just define a `ui.cursor`.